tests/catalog/test_base.py

```diff
@@ -355,8 +162,6 @@ def test_create_table_pyarrow_schema(catalog: InMemoryCatalog, pyarrow_schema_si
     table = catalog.create_table(
         identifier=TEST_TABLE_IDENTIFIER,
         schema=pyarrow_schema_simple_without_ids,
-        location=TEST_TABLE_LOCATION,
-        partition_spec=TEST_TABLE_PARTITION_SPEC,
```

Comment:
@syun64 FYI, I realized that the `TEST_TABLE_PARTITION_SPEC` here breaks this test. 
```
TEST_TABLE_PARTITION_SPEC = PartitionSpec(PartitionField(name="x", transform=IdentityTransform(), source_id=1, field_id=1000))
```
The partition field's `source_id` here is `1`, but in `create_table` the `schema`'s `field_id`s are all `-1` due to `_convert_schema_if_needed` 

So `assign_fresh_partition_spec_ids` fails

https://github.com/apache/iceberg-python/blob/102e043b182a0b7bf4c9f66c5d77b24eedbd3766/pyiceberg/partitioning.py#L203-L204

Reply:
Created https://github.com/apache/iceberg-python/issues/338 to track this issue

Reply:
I agree with you @syun64 that for creating tables having to look up the IDs is not ideal. Probably that API has to be extended at some point.

But for the metadata (and also how Iceberg internally tracks columns, since names can change; IDs not), we need to track it by ID. I'm in doubt if assigning `-1` was the best idea because that will give you a table that you cannot work with. Thanks for creating the issue, and let's continue there.

Reply:
Sounds good @Fokko 👍 and thanks again for flagging this @kevinjqliu !